src/features/copilotTools.ts

```diff
@@ -36,55 +50,181 @@ export class GetPackagesTool implements LanguageModelTool<IGetActiveFile> {
+            envInfo.runCommand = run.executable + (run.args && run.args.length > 0 ? ` ${run.args.join(' ')}` : '');
+            // TODO: check if this is the right way to get type
+            envInfo.type = environment.envId.managerId.split(':')[1];
+            envInfo.version = environment.version;
+
+            // does this need to be refreshed prior to returning to get any new packages?
             await this.api.refreshPackages(environment);
             const installedPackages = await this.api.getPackages(environment);
-
-            let resultMessage: string;
             if (!installedPackages || installedPackages.length === 0) {
-                resultMessage = 'No packages are installed in the current environment.';
+                envInfo.packages = [];
             } else {
-                const packageNames = installedPackages
-                    .map((pkg) => (pkg.version ? `${pkg.name} (${pkg.version})` : pkg.name))
-                    .join(', ');
-                resultMessage = 'The packages installed in the current environment are as follows:\n' + packageNames;
+                envInfo.packages = installedPackages.map((pkg) =>
+                    pkg.version ? `${pkg.name} (${pkg.version})` : pkg.name,
+                );
             }
 
-            const textPart = new LanguageModelTextPart(resultMessage || '');
+            // format and return
+            const textPart = BuildEnvironmentInfoContent(envInfo);
             deferredReturn.resolve({ content: [textPart] });
         } catch (error) {
-            const errorMessage: string = `An error occurred while fetching packages: ${error}`;
+            const errorMessage: string = `An error occurred while fetching environment information: ${error}`;
             deferredReturn.resolve({ content: [new LanguageModelTextPart(errorMessage)] } as LanguageModelToolResult);
         }
         return deferredReturn.promise;
     }
-
     /**
      * Prepares the invocation of the tool.
      * @param _options - The preparation options.
      * @param _token - The cancellation token.
      * @returns The prepared tool invocation.
      */
     async prepareInvocation?(
-        _options: LanguageModelToolInvocationPrepareOptions<IGetActiveFile>,
+        _options: LanguageModelToolInvocationPrepareOptions<IResourceReference>,
         _token: CancellationToken,
     ): Promise<PreparedToolInvocation> {
-        const message = 'Preparing to fetch the list of installed Python packages...';
+        const message = 'Preparing to fetch Python environment information...';
+        return {
+            invocationMessage: message,
+        };
+    }
+}
+
+function BuildEnvironmentInfoContent(envInfo: EnvironmentInfo): LanguageModelTextPart {
+    // Create a formatted string that looks like JSON but preserves comments
+    const content = `{
+  // type of python environment; sys means it is the system python
```

Comment:
The classification for system pythons is `system`. we don't use `sys`. I think the description here should change based on `system` vs other Pythons. We need to say something like this:

if type is system then use this description: "System pythons are pythons that ship with the OS or are installed globally. These python installs may be used by the OS for running services and core functionality. Confirm with the user before installing packages into this environment, as it can lead to issues with any services on the OS"

if type is not system: "This environment is managed by {type} environment manager. Use the install tool to install packages into this environment."